runtime/auth/identity-api/common/src/aws/smithy/kotlin/runtime/identity/Identity.kt

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package aws.smithy.kotlin.runtime.identity
+
+import aws.smithy.kotlin.runtime.time.Instant
+import aws.smithy.kotlin.runtime.util.Attributes
+
+/**
+ * Uniquely-distinguishing properties which identify an actor
+ */
+public interface Identity {
+    /**
+     * The time after which this identity will no longer be valid. If this is null, an expiration time
+     * is not known (but the identity may still expire at some point in the future).
+     */
+    public val expiration: Instant?
+
+    /**
+     * Typed property bag of attributes associated with this identity. Common attribute keys are defined in
+     * [IdentityAttributes].
+     */
+    public val attributes: Attributes
```

Comment:
Question: Should we make this nullable? Not every identity will carry attributes and this forces implementations to create empty property bags/go through lazy.

Reply:
I dislike nullable collection references. It begs the question: what's the difference between null and empty? I'd just as soon leave it empty for identities which carry no attributes.

A larger question for me is should the attribute collection be mutable? Everything else about identities/credentials are currently immutable so it seems weird to also carry a mutable collection but maybe there's a use case I'm not aware of.

Reply:
> what's the difference between null and empty?

In this case it's the difference is an allocation, there is no empty concept because `Attributes` is only mutable.

> A larger question for me is should the attribute collection be mutable? 

This would be a larger change to middleware/interceptors. I'm not sure what the fallout would be.